processing/src/main/java/org/apache/carbondata/processing/util/CarbonDataProcessorUtil.java

```diff
@@ -522,4 +522,16 @@ public static String checkAndCreateCarbonStoreLocation(String factStoreLocation,
     return aggType;
   }
 
+  /**
+   * the method prepares and return the message mentioning the reason of badrecord
+   *
+   * @param columnName
+   * @param dataType
+   * @return
+   */
+  public static String prepareFailureReason(String columnName, DataType dataType) {
```

Comment:
@when number of bad records are more then for each record prepare this message will be very costly as we know the data type and column name, if bad record is enabled can we prepare this message initially and store it in some map and whenever any bad record will come we can get this message from map based on column name. For each bad record prepare this message will hit data loading speed 
@kumarvishal09 what if bad record logger is enabled but no bad record present in the data.
and number of columns are bigger like 2000, is preparing map will be helpfull?